### DISCUSSION TITLE
Error importing pyserini.dsearch

### DISCUSSION THREAD

**Discussion by joaocp98662** [castorini/pyserini#1160] [Q&A] [answered]
Hello,

In my local machine everything works fine, with my environment. But when I try to run it in my lab machines (I don't have superuser permissions in this machines), after setting up everything, I get this error:

File "dense_retrieval.py", line 10, in <module>
    from pyserini.dsearch import SimpleDenseSearcher, AutoQueryEncoder
  File "/cfs/home/u121171/.local/lib/python3.6/site-packages/pyserini/dsearch/__init__.py", line 17, in <module>
    from pyserini.search.faiss import DenseSearchResult, PRFDenseSearchResult,  QueryEncoder, \
  File "/cfs/home/u121171/.local/lib/python3.6/site-packages/pyserini/search/__init__.py", line 17, in <module>
    from .lucene import JQuery, JDisjunctionMaxQueryGenerator, get_topics,\
  File "/cfs/home/u121171/.local/lib/python3.6/site-packages/pyserini/search/lucene/__init__.py", line 17, in <module>
    from ._base import JQuery, JDisjunctionMaxQueryGenerator, get_topics, \
  File "/cfs/home/u121171/.local/lib/python3.6/site-packages/pyserini/search/lucene/_base.py", line 34, in <module>
    JQrels = autoclass('io.anserini.eval.Qrels')
  File "/cfs/home/u121171/.local/lib/python3.6/site-packages/jnius/reflect.py", line 211, in autoclass
    c = find_javaclass(clsname)
  File "jnius/jnius_export_func.pxi", line 26, in jnius.find_javaclass
  File "jnius/jnius_utils.pxi", line 91, in jnius.check_exception
jnius.JavaException: JVM exception occurred: io/anserini/eval/Qrels has been compiled by a more recent version of the Java Runtime (class file version 55.0), this version of the Java Runtime only recognizes class file versions up to 52.0 java.lang.UnsupportedClassVersionError

my java version:

java -version
openjdk version "11.0.11" 2021-04-20
OpenJDK Runtime Environment (build 11.0.11+9-suse-3.56.1-x8664)
OpenJDK 64-Bit Server VM (build 11.0.11+9-suse-3.56.1-x8664, mixed mode)

I know the problem is on my side, but can someone help me please?

Thank you all,

Regards

**Comment by lintool** [comment, marked as answer]
It's a JVM version issue...

```
52 = Java 8
53 = Java 9
54 = Java 10
55 = Java 11
```

You'll need to find some way to run the right JVM version on your lab machine... this depends on your lab policy, which isn't something we can really help with...

**Comment by zhenduow** [comment]
Have the same problem. How to make sure I run it with Java 11?

**Comment by Xnhyacinth** [comment]
I have run the right JVM version.

java -version:
openjdk version "11.0.21" 2023-10-17
OpenJDK Runtime Environment (build 11.0.21+9-post-Ubuntu-0ubuntu122.04)
OpenJDK 64-Bit Server VM (build 11.0.21+9-post-Ubuntu-0ubuntu122.04, mixed mode, sharing)

However, the error is changed:
jnius.JavaException: JVM exception occurred: io/anserini/analysis/DefaultEnglishAnalyzer has been compiled by a more recent version of the Java Runtime (class file version 65.0), this version of the Java Runtime only recognizes class file versions up to 55.0 java.lang.UnsupportedClassVersionError.

How can I solve it.

**Reply by lintool** [reply]
Hi @Xnhyacinth - please see my response here:
https://github.com/castorini/pyserini/issues/383#issuecomment-2212548193
